imports/plugins/core/graphql/server/schemas/cart.graphql

```diff
@@ -0,0 +1,346 @@
+}
+
+"A connection edge in which each node is a `CartItem` object"
+type CartItemEdge implements NodeEdge {
+  cursor: ConnectionCursor!
+  node: CartItem
+}
+
+"""
+A single item in a cart. The item contains information about an intended purchase.
+"""
+type CartItem implements Node {
+  "The cart item ID"
+  _id: ID!
+
+  "The date and time at which this item was added to the associated cart. If an item is added, removed, and then added again, this will reflect the most recent addition."
+  addedAt: DateTime! # Should this be "createdAt" ?
+
+  "The date and time at which the cart item was created. If an item is added, removed, and then added again, the original item is destroyed and this field will reflect the time it was created for the most recent addition."
+  createdAt: DateTime!
+
+  "Arbitrary additional metadata about this cart item."
+  metafields: [Metafield]
+
+  "Packing information such as item weight, height, length, and depth."
+  parcel: ShippingParcel
+
+  "The current price of the item"
+  price: Float!
+
+  "The currency for the `price` and `priceWhenAdded` fields"
+  priceCurrency: Currency!
+
+  # This will permit us to perform calculations based on how the price has changed since
+  # the item was added to the cart.
+  "The price at which this item was listed when it was added to the cart"
+  priceWhenAdded: Float!
+
+  "The product and chosen options"
+  productConfiguration: ProductConfiguration!
+
+  "The quantity of this item that has been added to the cart. This must be a positive integer. Remove this `CartItem` from it's associated cart if you want `0` of this item."
+  quantity: Int!
+
+  # QUESTION: This implies that cart items can be from different shops.
+  # Is it possible that certain shops share a cart but other shops are not
+  # included in that sharing? How is that defined? Probably not applicable on
+  # the cart item but at the cart itself.
+  #
+  # TODO: shopId is not currently required but it should be made required to
+  # avoid issues on cart sharing reconfiguration.
+  "The shop associated with this cart item."
+  shop: Shop
+
+  # TODO: What is this? Simply denormalization? Can this attribute be fetched
+  # directly from the product now in GraphQL?
+  # Should this be embedded within the productConfiguration
+  "The type of product, used to display cart items differently"
+  type: String
+
+  "The date and time at which this item was last updated"
+  updatedAt: DateTime!
+}
+
+"Product configuration data"
+type ProductConfiguration {
+  # This is Product ID, but should not be used to couple the cart item or product configuration to the Product
+  # We're adding Product to the cart which is why we're not using the CatalogProductId here
+  "The Product ID"
+  productId: ID!
+
+  "The ProductVariant ID"
+  productVariantId: ID!
+
+  "The product's slug"
+  productSlug: String
+
+  "A title for use in cart/orders that conveys the selected product's title + chosen options"
+  title: String!
```

Comment:
Let's do that. Move everything to `CartItem` except `productId` and `productVariantId`. We'll keep `ProductConfiguration` as the grouper for those, which would later have the selected configurable attribute values on it, too.

Then I think `CartItem` can have a _separate_ prop, `attributes`, which will be all the configurable attributes and their chosen values. We can do that now because we can map them from the current variant/option structure into a flat attribute structure.

@spencern If this isn't clear, we can make the changes pretty quickly if we pair them.